pynitrokey/fido2/client.py

```diff
@@ -67,19 +67,22 @@ def reboot(
             pass
 
     def find_device(self, dev=None, solo_serial: str = None):
+        devices = []
         if dev is None:
-            devices = list(CtapHidDevice.list_devices())
             if solo_serial is not None:
                 if solo_serial.startswith("device="):
                     solo_serial = solo_serial.split("=")[1]
                     dev = open_device(solo_serial)
                 else:
+                    devices = list(CtapHidDevice.list_devices())
```

Comment:
This still connects to all devices if a serial number is given.  Instead we could list the descriptors (`fido2.hid.list_descriptors`), check the serial number and the connect only to the matching device(s).

Reply:
Sounds good! Leaving open until corrected then.

Reply:
I think fixing it for the HID path case is the more important part, so we can also merge this PR now and open a ticket to keep track of the serial number case.